libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AnalyticsPublishingWorker.kt

```diff
@@ -27,26 +27,35 @@
 package com.salesforce.androidsdk.analytics
 
 import android.content.Context
+import androidx.work.Constraints
 import androidx.work.ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE
 import androidx.work.ListenableWorker.Result.success
-import androidx.work.PeriodicWorkRequest.Builder
+import androidx.work.NetworkType.CONNECTED
+import androidx.work.OneTimeWorkRequest
+import androidx.work.PeriodicWorkRequest
 import androidx.work.WorkManager.getInstance
 import androidx.work.Worker
 import androidx.work.WorkerParameters
 import com.salesforce.androidsdk.accounts.UserAccountManager
-import com.salesforce.androidsdk.analytics.AnalyticsPublishingWorker.Companion.reEnqueueAnalyticsPublishPeriodicWorkRequest
+import com.salesforce.androidsdk.analytics.AnalyticsPublishingWorker.Companion.enqueueAnalyticsPublishWorkRequest
 import java.util.concurrent.TimeUnit.HOURS
 
 /**
  * An Android background tasks worker which publishes stored analytics.
  * This class is intended to be instantiated by the background tasks work
  * manager.
  *
- * Use [reEnqueueAnalyticsPublishPeriodicWorkRequest] to enqueue a analytics
- * publish periodic work request.  Previous requests will be cancelled.
+ * [enqueueAnalyticsPublishWorkRequest] is used internally by the Salesforce
+ * Mobile SDK to enqueue analytics publish work requests to match the current
+ * analytics publishing configuration. Only one request may be active at a time.
+ * Only the Salesforce Mobile SDK should call this method as it is not intended
+ * for public use.
  *
  * @param context The Android context provided by the work manager
  * @param workerParams The worker parameters provided by the work manager
+ * @see [SalesforceAnalyticsManager.setPublishOnceTimeOnAppBackgroundEnabled]
+ * @see [SalesforceAnalyticsManager.setPublishPeriodicallyOnFrequencyEnabled]
+ * @see [SalesforceAnalyticsManager.setPublishPeriodicallyFrequencyHours]
```

Comment:
I wonder if we should support three modes (and therefore have three methods):
- no publishing: turnOffPublishing()
- publish on background: publishOnceOnAppBackground()
- publish periodically with period x: publishPeriodically(frequencyHours)

Right now it looks like you could publish on background and periodically, or you could publish periodically without specifying a frequency.